File: Problems/967_Numbers_With_Same_Consecutive_Differences.cpp
class Solution {
public:
    vector<int> numsSameConsecDiff(int n, int k) {
        vector<int> res;
        for (int i = 1; i < 10; ++i) {
            helper(n, k, res, i, 1);
        }
        return res;
    }
    void helper(int n, int k, vector<int> &res, int num, int digit) {
        if (digit == n) {
            res.push_back(num);
            return;
        }
        int tailNum = num % 10;
        int tmp = num;
        if (tailNum + k < 10) {
            tmp *= 10;
            tmp += tailNum + k;
            helper(n, k, res, tmp, digit + 1);
        }
        if (!k) return;
        tmp = num;
        if (tailNum - k >= 0) {
            tmp *= 10;
            tmp += tailNum - k; 
            helper(n, k, res, tmp, digit + 1);
        }
    }
};

File: Problems/967_Numbers_With_Same_Consecutive_Differences.py
class Solution:
    def numsSameConsecDiff(self, n: int, k: int) -> List[int]:
        res = []
        
        def helper(num, digit):
            if digit == n:
                res.append(num)
                return
            lastDigit = num % 10
            tmp = num
            if lastDigit + k < 10:
                tmp *= 10
                tmp += lastDigit + k
                helper(tmp, digit + 1)
            if not k:
                return
            tmp = num
            if lastDigit - k >= 0:
                tmp *= 10
                tmp += lastDigit - k
                helper(tmp, digit + 1)
        
        for i in range(1, 10):
            helper(i, 1)
        
        return res